agenta-web/src/components/TestSetTable/TestsetTable.tsx

```diff
@@ -194,12 +189,124 @@ const TestsetTable: React.FC<testsetTableProps> = ({mode}) => {
         })
 
         setColumnDefs(newColumnDefs)
+
         setRowData(newRowData)
         if (gridRef.current) {
             gridRef.current.setColumnDefs(newColumnDefs)
         }
     }
 
+    const HeaderComponent = (params: any) => {
+        const {eGridHeader} = params
+        const [scopedInputValues, setScopedInputValues] = useState(
+            columnDefs.filter((colDef) => colDef.field !== "").map((col) => col.field),
+        )
+        const [index, setIndex] = useState(eGridHeader.attributes[4].nodeValue - 2)
+
+        const [displayName, setDisplayName] = useState(
+            inputValues[eGridHeader.attributes[4].nodeValue - 2],
+        )
+
+        const [error, setError] = useState<string | undefined>(undefined)
+
+        const [isEditModalOpen, setIsEditModalOpen] = useState<boolean>(false)
+        const handleOpenIsEditModalOpen = () => {
+            setIsEditModalOpen(true)
+        }
+
+        const handleCloseIsEditModalOpen = () => {
+            setScopedInputValues(inputValues)
+            setIsEditModalOpen(false)
+        }
+
+        const handleSave = () => {
+            if (inputValues.includes(scopedInputValues[index])) {
+                setError("Column name already exist!")
```

Comment:
I would recommend using antd `message.error()` function here and avoid maintaining an error state

Reply:
Will do